fastify docs (#233)
Genez-io/genezio-documentation: docs/frameworks/fastifyjs.md

---
description: Learn how to deploy a Fastify.js application with Genezio.
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

> # Fastify.js

Reviewer: From their landing page[1] it doesn't seem that `Fastify` has the suffix `.js` .

I would remove .`js` or `js` terminations from this page to align with their branding.

https://fastify.dev/

<head>
    <title>Fastify.js | Genezio Documentation</title>
</head>

Fastify.js is a popular Node.js web application framework that simplifies the development of server-side applications. It provides a robust set of features for building web servers and APIs.

> <!-- TODO: change to fastify template -->

Reviewer: ```suggestion
```

:::tip
Get started in no time with the [Fastify.js template](https://github.com/Genez-io/fastify-getting-started).
:::

# Deployment

> Learn how to deploy an existing Fastify.js app using Genezio, a serverless deployment platform that simplifies app management and reduces costs

Reviewer: ```suggestion
Learn how to deploy an existing Fastify.js app using Genezio, a serverless deployment platform that simplifies app management and reduces costs.
```

## Prerequisites

### 1. Install genezio

Use your preferred package manager to install Genezio:

<Tabs>
  <TabItem className="tab-item" value="npm" label="npm">
<div id="step1-install-npm">
  ```
  npm install genezio -g
  ```
  </div>
  </TabItem>
  <TabItem className="tab-item" value="pnpm" label="pnpm">
  <div id="step1-install-pnpm">
  ```
  pnpm add -g genezio
  ```
  </div>
  </TabItem>
  <TabItem  className="tab-item" value="yarn" label="yarn">
  <div id="step1-install-yarn">
  ```
  yarn add global genezio
  ```
  </div>
  </TabItem>
</Tabs>

### 2. Ensure you have an Fastify.js App

If you don't have an Fastify.js app, you can create one using the following steps:

<details>
  <summary>**Create a Hello World Fastify.js App**</summary>

> ### 1. Initialize a New Node.js Project

Reviewer: ```suggestion
<h3> 1. Initialize a New Node.js Project </h3>
```

Reviewer: This looks a bit weird in the table of contents on the right side because docusaurus doesn't detect that this is an expandable section.

To fix this we should hide this headers from the ToC.

I researched a bit and we can do that by [changing the markdown headers (`###`) to html header tags.](https://docusaurus.io/docs/next/markdown-features/toc#customizing-table-of-contents-generation)

Please to this change for all the headers in the expandable section.

<img width="297" alt="image" src="https://github.com/user-attachments/assets/e5273187-430e-408c-abca-f411a52c9686">

:::info

1. You might need to replace the `entry` field with the name of your main application file.
2. You might need to replace the `path` field with the path relative at **genezio.yaml** file.
3. This example configuration works if **genezio.yaml** is in the same directory as your main application file.
   :::

```yaml title="genezio.yaml"
# The name of the project.
name: fastify-app
# The region where the project is deployed. Available regions: us-east-1, eu-central-1
region: us-east-1
# The version of the Genezio YAML configuration to parse.
yamlVersion: 2
backend:
  # The root directory of the backend.
  path: ./
  # Information about the backend's programming language.
  language:
    # The name of the programming language.
    name: js
    # The package manager used by the backend.
    packageManager: npm
  # Information about the backend's functions.
  functions:
    # The name (label) of the function.
    - name: hello-world-fastify-app-function
      # The path to the function's code.
      path: ./
      # The name of the function handler
      handler: handler
      # The entry point for the function.
      entry: app.mjs

>       # The compatibility of the function handler.
>       type: aws

Reviewer: ```suggestion
```

This field is not required and it might confuse the user. You can remove it.

```

This configuration file specifies the project name, deployment region, and details about the backend.

## 4. Test Your App Locally

Before deploying your app, you can test it locally to ensure it's working correctly.

Run the following command in your terminal:

```bash
NODE_ENV=dev node app.mjs
```

## 5. Deploy your project

Finally, deploy your project. A browser window will open, and you will be prompted to log in to your Genezio account and authorize the CLI to make the deployment.
Run the following command in your terminal:

```bash
genezio deploy
```

If your application use environment variables, you can deploy them using the following command:

```bash
genezio deploy --env <path-to-your-env-file>
```

:::info

> You need to deploy your environment variables single time.

Reviewer: ```suggestion
You need to deploy your environment variables only once.
```

After that, you can deploy your project without the `--env` flag.
:::
For more information about environment variables, you can check the [official documentation](/docs/project-structure/backend-environment-variables.md).

## See your app in Genezio Dashboard

After deploying your application, you can test it to ensure it's running correctly. To verify that your Fastify.js app is working, open a web browser and navigate to the URL provided for your deployed function.

This URL can be found in the deployment output under the `Functions Deployed` section.

Additionally, you can monitor and manage your app through the [Genezio App Dashboard](https://app.genez.io/dashboard). The dashboard URL, also provided after deployment, allows you to access comprehensive views of your project's status and logs.

You can find this URL in the deployment output under the `App Dashboard URL` section.

## Frequently Asked Questions

### Do I Need to Modify My Fastify App Code?

> Your fastify code will only require the above [adjustments](#deployment-guide), as long as it is written in a **stateless** manner. [Read more](https://stackoverflow.com/questions/5329618/stateless-vs-stateful)

Reviewer: ```suggestion
Your Fastify project will only require the above [adjustments](#deployment-guide), as long as it is written in a **stateless** manner. To find out more about [stateless vs. stateful follow this link](https://stackoverflow.com/questions/5329618/stateless-vs-stateful)
```
